api/server/image.go

```diff
@@ -112,7 +113,7 @@ func (s *Server) postImagesCreate(ctx context.Context, w http.ResponseWriter, r
 			OutStream:   output,
 		}
 
-		err = s.daemon.Repositories().Pull(image, tag, imagePullConfig)
+		err = s.daemon.Repositories().Pull(image, tag, imagePullConfig, dryRun != "")
```

Comment:
This would turn on the "dryRun" mode even if `dryRun` was set to false in the request.